bnr-utils/nz_azConnector/nz_azConnector.go

```diff
@@ -384,25 +388,37 @@ func main() {
     // parse input args
     parseArgs(&conn, &backupinfo, &othargs)
     flag.Parse()
+    
+    requiredFlags := []string{ "db","dir","npshost","storage-account","key","container",
+    "upload","backupset","uniqueid","logfiledir","streams","paralleljobs","blocksize"}
 
-    if flag.NFlag() == 0 {
-        fmt.Println("No arguments passed to nz_azConnector. Below is the list of valid args:")
+    for _, flagName := range requiredFlags {
+        if flag.Lookup(flagName).Value.String() == "" {
+        fmt.Printf("Error: %s Invalid flag\n", flagName)
         flag.PrintDefaults()
+        os.Exit(1)
+        }

```

Comment:
Wouldn't it be better to first look at env variables and if not set then error out for required flags

```go
requiredFlags := map[string]string { 
    "db" : "NZ_DATABASE", 
    "dir" : "",
    "npshost" : "NZ_HOST",
    // and so on for the rest.
};

ensureDefault := func(f string, env string) bool {
    val := flag.Lookup(f).Value
    if Value.String() == "" && env != " {
        val.Set(os.Getenv(env));
    }
    return val.String() != "";
}

for f, env := range requiredFlags {
    if !ensureDefault(f, env) {
        fmt.Fprintf(stderr, "-%s is required", f)
        os.Exit(1)
    }
}
```

Reply:
ok I will work on this